### ISSUE TITLE
Use stdout to parse airodump output?

### ISSUE THREAD

**Issue by llazzaro** [XayOn/pyrcrack#2]
Hi,

I'm using a fork of this project where I change how to parse the airodump-ng, I'm using stdout and it seems to update more fast and also avoids the usage of disk for writting the csv. (and all the code behind reading the file is error prone).
I also had to patch the write parameter since change it will cause issues.

Let me know what do you think, I could try to do a pull request.

**Comment by XayOn** [comment]
Hi there Leonardo!

I'd like to merge your commit 8e577d74919b0aef7d19355556da12c68efe34b1, 195465f0fe2e4b1f45adb5024d64fec5e39b0293 and 87d2b40ba75a26fd97bf03d2f1a7ad982fef4dc6
if you could send a PR for those I'd really appreciate it.

I've seen you added a test (I really really need to do more tests) for the stdout part. Wouldn't it be better to just add the testcases where the current code fails, and fix the current code?

Don't get me wrong, it's not a final decision, it's just that parsing output it's really CPU demanding.
I've seen people in the real world almost burning pieces of hardware where they had a process parsing aircrack-ng's output!

The reason behind that is that aircrack-ng makes a while True loop that outputs in screen and then clears it while, for file writing, it does a write every 5 seconds. I believe that processing data from stdout is less reliable, and can cause more problems. It also adds an extra layer escaping characters.

I believe your problem might come from this:

> From python3.4's open() documentation: In text mode, if encoding is not specified the encoding used is platform dependent: locale.getpreferredencoding(False) is called to get the current locale encoding

This means that if your locale isn't configured to use utf-8, pyrcrack will fail. And that is undoubtedly a bug in pyrcrack. If you could try to force it to utf-8 that'd be really nice (and a test would be really nice, but I'm not sure how to test this).

Aircrack-ng seems to open files in wb+, wich would lead to thinking that pyrcrack should do the same, but unfortunately csv module only supports iterators that return strings, not bytes. Yet I believe using utf-8 by default should be enough. The CSV files are what aircrack-ng provides and should be the same always, so I believe checking for it each time should not be needed.

Thank you!

**Comment by llazzaro** [comment]
@XayOn Good point about high cpu usage on parsing the stdout. I just finished the code for the stdout parsing, I will do some tests...but I will probably change my mind and roll back to csv parsing.
I will do the pr for those commits and if possible add some tests in the future.

**Comment by llazzaro** [comment]
You were right about parsing stdout. To improve performance I used stdin to send "pause" to the airodump (I'm not sure if it pauses the capture also).
However the csv is much more cleaner and I will stay with it.
I added the write-interval parameter (I dind't push it yet), for me 5 seconds could be a lot of time to react. I can live with 1 second.
The only thing I did in my last commits was to break the data type returned, so I think I wont do a pull request since it will break your applications.

I did a pull request with the requested commits.
